File: app/src/main/java/com/ivanovsky/passnotes/ui/groups/GroupsActivity.java
package com.ivanovsky.passnotes.ui.groups;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.MenuItem;

import com.ivanovsky.passnotes.App;
import com.ivanovsky.passnotes.R;
import com.ivanovsky.passnotes.data.safedb.EncryptedDatabaseProvider;
import com.ivanovsky.passnotes.databinding.CoreBaseActivityBinding;
import com.ivanovsky.passnotes.ui.core.BaseActivity;

import javax.inject.Inject;

public class GroupsActivity extends BaseActivity {

	@Inject
	EncryptedDatabaseProvider dbProvider;

	public static Intent createIntent(Context context) {
		return new Intent(context, GroupsActivity.class);
	}

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		App.getDaggerComponent().inject(this);

		CoreBaseActivityBinding binding = DataBindingUtil.setContentView(this, R.layout.core_base_activity);

		setSupportActionBar(binding.toolBar);
		getCurrentActionBar().setTitle(R.string.groups);
		getCurrentActionBar().setDisplayHomeAsUpEnabled(true);

		GroupsFragment fragment = GroupsFragment.newInstance();

		getSupportFragmentManager().beginTransaction()
				.replace(R.id.fragment_container, fragment)
				.commit();

		GroupsContract.Presenter presenter = new GroupsPresenter(this, fragment);
		fragment.setPresenter(presenter);
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		if (item.getItemId() == android.R.id.home) {

			if (dbProvider.isOpened()) {
				dbProvider.close();
			}

			onBackPressed();
			return true;
		} else {
			return super.onOptionsItemSelected(item);
		}
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/ui/groups/GroupsActivity.kt
package com.ivanovsky.passnotes.ui.groups

import android.content.Context
import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.MenuItem
import com.ivanovsky.passnotes.App
import com.ivanovsky.passnotes.R
import com.ivanovsky.passnotes.data.repository.EncryptedDatabaseRepository
import com.ivanovsky.passnotes.databinding.CoreBaseActivityBinding
import com.ivanovsky.passnotes.injection.Injector
import com.ivanovsky.passnotes.presentation.core.BaseActivity
import javax.inject.Inject

class GroupsActivity : BaseActivity() {

	companion object {

		fun createStartIntent(context: Context): Intent {
			return Intent(context, GroupsActivity::class.java)
		}
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		Injector.getInstance().appComponent.inject(this)

		val binding = DataBindingUtil.setContentView<CoreBaseActivityBinding>(this,
				R.layout.core_base_activity)

		setSupportActionBar(binding.toolBar)
		currentActionBar.title = getString(R.string.groups)
		currentActionBar.setDisplayHomeAsUpEnabled(true)

		val fragment = GroupsFragment.newInstance()

		supportFragmentManager.beginTransaction()
				.replace(R.id.fragment_container, fragment)
				.commit()

		val presenter = GroupsPresenter(this, fragment)
		fragment.setPresenter(presenter)
	}

	override fun onOptionsItemSelected(item: MenuItem?): Boolean {
		if (item?.itemId == android.R.id.home) {
			Injector.getInstance().releaseEncryptedDatabaseComponent()

			finish()

			return true
		} else {
			return super.onOptionsItemSelected(item)
		}
	}
}
